### PR TITLE
feat(auth): add documentation for Play Game Sign-In

### PR DESCRIPTION
**Pull request by Lyokone**
## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change.*

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: docs/auth/federated-auth.md [hunk 1]
```diff
@@ -91,6 +91,39 @@ Ensure the "Google" sign-in provider is enabled on the [Firebase Console](https:
   }
   ```
 
+## Game Auth Provider
+Firebase is compatible with specific game provider on Android (Play Games Sign-In) and iOS (Game Center).
```
Comment (russellwheatley) [inline docs/auth/federated-auth.md:95]:
```suggestion

Firebase is compatible with specific game provider on Android (Play Games Sign-In) and iOS (Game Center).
```

File: docs/auth/federated-auth.md [hunk 2]
```diff
@@ -91,6 +91,39 @@ Ensure the "Google" sign-in provider is enabled on the [Firebase Console](https:
...
+
+  Not supported yet.
+
+* {Android}
+
+  Follow the instructions for Google setup on Android, then configure 
+  [Play Games services with your Firebase app information](https://firebase.google.com/docs/auth/android/play-games#configure-play-games-with-firebase-info).
+
+  You can connect your user using:
+  ```dart
```
Comment (russellwheatley) [inline docs/auth/federated-auth.md:107]:
```suggestion

 ```dart
```

File: docs/auth/federated-auth.md [hunk 3]
```diff
@@ -91,6 +91,39 @@ Ensure the "Google" sign-in provider is enabled on the [Firebase Console](https:
...
+* {iOS}
+
+  Not supported yet.
+
+* {Android}
+
+  Follow the instructions for Google setup on Android, then configure 
+  [Play Games services with your Firebase app information](https://firebase.google.com/docs/auth/android/play-games#configure-play-games-with-firebase-info).
+
+  You can connect your user using:
```
Comment (russellwheatley) [inline docs/auth/federated-auth.md:106]:
```suggestion
  The following will trigger the sign-in flow, create a new credential and sign in the user:
```

**Review by russellwheatley — CHANGES_REQUESTED** [review]

**Comment by russellwheatley** [comment]
Hey @kevinthecheung, could you let us know if this is acceptable?

File: docs/auth/federated-auth.md [hunk 4]
```diff
@@ -91,6 +91,41 @@ Ensure the "Google" sign-in provider is enabled on the [Firebase Console](https:
   }
   ```
 
+## Game Auth Provider
+
+Firebase is compatible with specific game provider on Android (Play Games Sign-In) and iOS (Game Center).
```
Comment (kevinthecheung) [inline docs/auth/federated-auth.md:96]: If Game Center isn't supported yet, how about just calling the section "Play Games sign-in" and removing the iOS tab?

**Review by kevinthecheung — CHANGES_REQUESTED** [review]

File: docs/auth/federated-auth.md (hunk 4 above)
Comment (Lyokone) [reply docs/auth/federated-auth.md]: I agree with you and I have made the requested change. I was expecting Game Center integration to be easier

**Comment by Lyokone** [comment]
@russellwheatley @kevinthecheung It should be all good now :)

**Review by russellwheatley — APPROVED** [review]

**Pull request merged**